test/Microsoft.Data.Sqlite.Tests/SqliteConnectionTest.cs

```diff
@@ -705,6 +705,40 @@ public void CreateFunction_works_when_parameter_empty_blob()
             }
         }
 
+        [Fact]
+        public void CreateFunction_is_non_deterministic()
+        {
+            using (var connection = new SqliteConnection("Data Source=:memory:"))
+            {
+                connection.Open();
+
+                connection.ExecuteNonQuery("CREATE TABLE Data (Value); INSERT INTO Data VALUES (0);");
+                connection.CreateFunction("test", (double x) => x);
+                
+                var ex = Assert.Throws<SqliteException>(
+                    () => connection.ExecuteNonQuery("CREATE INDEX InvalidIndex ON Data (Value) WHERE test(Value) = 0;"));
+
+                Assert.Equal(
+                    Resources.SqliteNativeError(raw.SQLITE_ERROR, Resources.NonDeterministicFunctionProhibitedInIndex),
```

Comment:
Could you assert the extended error code instead of the message?

Reply:
Unfortunately there seems to be no specific extended error code for this error (ex.SqliteExtendedErrorCode is 1).

Reply:
Ok. Don't add a new resource, just assert the English text--SQLite isn't localized

Reply:
Done.